optaplanner-core/src/main/java/org/optaplanner/core/api/stat/OptaPlannerMetrics.java

```diff
@@ -0,0 +1,88 @@
+
+import io.micrometer.core.instrument.FunctionCounter;
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.binder.MeterBinder;
+
+public class OptaPlannerMetrics implements MeterBinder {
+
+    private final Solver<?> solver;
+    private final Iterable<Tag> tags;
+    private final ScoreCalculationSpeedSubSingleStatisticListener scoreCalculationSpeedSubSingleStatisticListener;
+
+    public OptaPlannerMetrics(Solver<?> solver, Iterable<Tag> tags) {
+        this.tags = tags;
+        this.solver = solver;
+        // if (solver instanceof AbstractSolver) {
+        //     throw new IllegalArgumentException("Solver (" + solver + ") does not extend (" + AbstractSolver.class
+        //             + ") and cannot be monitored. Maybe obtain a Solver from SolverFactory?");
+        // }
+        AbstractSolver<?> abstractSolver = (AbstractSolver<?>) solver;
+        scoreCalculationSpeedSubSingleStatisticListener = new ScoreCalculationSpeedSubSingleStatisticListener();
+
+        abstractSolver.addPhaseLifecycleListener(scoreCalculationSpeedSubSingleStatisticListener);
+    }
+
+    public static void monitor(MeterRegistry registry, Solver<?> solver, String... tags) {
+        new OptaPlannerMetrics(solver, Tags.of(tags)).bindTo(registry);
+    }
+
+    private <T> void counter(MeterRegistry registry, String name,
+            String description, T listener, ToDoubleFunction<T> statistic, String... extraTags) {
+        if (listener == null) {
+            return;
+        }
+        System.identityHashCode(solver);
+        FunctionCounter.builder(name, listener, statistic)
+                .tags(tags)
+                .tags(extraTags)
+                .description(description)
+                .register(registry);
+    }
+
+    private <T> void gauge(MeterRegistry registry, String name,
+            String description, T listener, ToDoubleFunction<T> statistic, String... extraTags) {
+        if (listener == null) {
+            return;
+        }
+        Gauge.builder(name, listener, statistic)
+                .description(description)
+                .tags(tags)
+                .tags(extraTags)
+                .register(registry);
+    }
+
+    @Override
+    public void bindTo(MeterRegistry registry) {
+        counter(registry, "optaplanner.solver.calculation-count", "Total calculations performed by the solver",
+                scoreCalculationSpeedSubSingleStatisticListener,
+                ScoreCalculationSpeedSubSingleStatisticListener::getScoreCalculationCount);
+        gauge(registry, "optaplanner.solver.calculation-speed", "The score calculation speed of the solver",
+                scoreCalculationSpeedSubSingleStatisticListener,
+                ScoreCalculationSpeedSubSingleStatisticListener::getScoreCalculationSpeed);
+    }
```

Comment:
How do I disable these if I don't want some of them?

These will have a non-trivial performance impact, when enabled. The users should be able to chose just the ones they need. (Alternatively, they should be able to exclude the ones they do not want.)

Reply:
Micrometer provides this functionality for us: https://micrometer.io/docs/concepts#_meter_filters

Reply:
I assumed it would. This solves the problem of disabling it by the users of core.
That said, we still need a way for benchmarker to do this, and if that was the same way, that would be nice.
I still think that solver config would be the way to go here - expose it on our level, as opposed to exposing Micrometer.